libfoundation/src/foundation-java-private.cpp

```diff
@@ -0,0 +1,1353 @@
+                                                       r_return);
+                else
+                    return __JavaJNINonVirtualMethodResult(t_instance,
+                                                           t_target_class,
+                                                           static_cast<jmethodID>(p_method_id),
+                                                           nullptr,
+                                                           t_return_type,
+                                                           r_return);
+        }
+        case MCJavaCallTypeGetter:
+        case MCJavaCallTypeSetter:
+        {
+            MCAssert(t_param_count > 0);
+            jobject t_instance = t_params[0].l;
+            
+            if (p_call_type == MCJavaCallTypeGetter)
+                return __JavaJNIGetFieldResult(t_instance,
+                                               static_cast<jfieldID>(p_method_id),
+                                               t_return_type,
+                                               r_return);
+            
+            MCJavaType t_field_type;
+            if (!__GetExpectedTypeCode(MCHandlerTypeInfoGetParameterType(p_signature, 1),
+                                       t_field_type))
+                return false;
+            
+            __JavaJNISetFieldResult(t_instance,
+                                    static_cast<jfieldID>(p_method_id),
+                                    &t_params[1],
+                                    t_field_type);
+        }
+        case MCJavaCallTypeStaticGetter:
+        case MCJavaCallTypeStaticSetter:
+        {
+            jclass t_target_class = s_env -> FindClass(*t_class_cstring);
+            if (p_call_type == MCJavaCallTypeStaticGetter)
+                return __JavaJNIGetStaticFieldResult(t_target_class,
+                                                     static_cast<jfieldID>(p_method_id),
+                                                     t_return_type,
+                                                     r_return);
+            
+            MCJavaType t_field_type;
+            if (!__GetExpectedTypeCode(MCHandlerTypeInfoGetParameterType(p_signature, 1),
+                                       t_field_type))
+                return false;
+            
+            __JavaJNISetStaticFieldResult(t_target_class,
+                                          static_cast<jfieldID>(p_method_id),
+                                          &t_params[0],
+                                          t_field_type);
+        }
+    }
+    
+    return false;
+}
+
+bool MCJavaPrivateObjectDescribe(MCValueRef p_value, MCStringRef &r_desc)
+{
+    MCJavaObjectRef t_obj = static_cast<MCJavaObjectRef>(p_value);
+    
+    MCAutoStringRef t_class_name;
+    if (!MCJavaPrivateGetJObjectClassName(t_obj, &t_class_name))
+        return false;
+    
+    return MCStringFormat(r_desc, "<java: %@>", *t_class_name);
```

Comment:
To be able to distinguish between different instances of the same Java class, I recommend including something unique (like the instance address) in the describe string.